includes/polling/os/awplus.inc.php

```diff
@@ -3,13 +3,30 @@
 //$hardware and $serial use snmp_getnext as the OID for these is not always fixed.
 //However, the first OID is the device baseboard.
 
-$hardware = snmp_getnext($device, "rscBoardName", "-OQv", "AT-RESOURCE-MIB");
+$data = snmp_getnext_multi($device, 'rscBoardName rscBoardSerialNumber', '-OQs', 'AT-RESOURCE-MIB');
+$hardware = $data['rscBoardName'];
+$serial = $data['rscBoardSerialNumber'];
 $version = snmp_get($device, "currSoftVersion.0", "-OQv", "AT-SETUP-MIB");
-$hostname = snmp_get($device, "sysName.0", "-OQv", "SNMPv2-MIB");
-$serial = snmp_getnext($device, "rscBoardSerialNumber", "-OQv", "AT-RESOURCE-MIB");
 
 // SBx8100 platform has line cards show up first in "rscBoardName" above.
+//Instead use sysObjectID.0
+
 if (strpos($hardware, 'SBx81') !== false) {
-    $hardware = snmp_get($device, "sysObjectID.0", "-OQvs", "SNMPv2-MIB:AT-PRODUCT-MIB");
-    $hardware = str_replace('at', '', $hardware);
+    $data_array = snmpwalk_cache_multi_oid($device, 'rscBoardName', $data_array, 'AT-RESOURCE-MIB', '-OUsb');
+    $data_array = snmpwalk_cache_multi_oid($device, 'rscBoardSerialNumber', $data_array, 'AT-RESOURCE-MIB', '-OUsb');
+
+    $hardware = snmp_translate($device['sysObjectID'], 'AT-PRODUCT-MIB', null, null, $device);
+    $hardware = str_replace('at', 'AT-', $hardware);
+
+// Features and Serial is set to Controller card 1.5
+    $features = $data_array['5.6']['rscBoardName'];
+    $serial = $data_array['5.6']['rscBoardSerialNumber'];
+    $serial = $serial. ' - CFC Bay 1.5';
```

Comment:
Last one for me on this, why append something to the end of the serial? The serial field should solely contain the actual serial number.

Reply:
@laf 

No worries. I tried to answer this in my last comment, but to provide some more context.

Essentially, we could have up to 24 serial numbers for 24 line cards, with 4 cards being controller cards in a 2x 12 bay Chassis stack.

Then there's also the serial number for the chassis shell, and serials for the PSUs etc. 

So as a user, I thought it was important to clarify *what* serial number we are actually reporting in LibreNMS. 

I mean, as a user which serial should I also care about? In almost all cases - The Controller Card. This is because all licensing is tied to it. 
 
So as a result, the serial number we most likely care about though is going to be either CFC Bay 1.5, or if not present then CFC Bay 1.6:

```
! Starting: Show system cmd 
! --------------------------------------------------------------------------------! 
! Stack member 1
! Board      ID   Bay     Board Name                        Rev   Serial number
! --------------------------------------------------------------------------------
! Chassis    374  Chassis AT-SBx8106                        B-0   R265E5008       
! Blade      353  Bay1    AT-SBx81XS6                       X8-0  A045624113500003
! Blade      317  Bay3    AT-SBx81GP24                      A-0   A04200a112800003
! Blade      352  Bay4    AT-SBx81GS24a                     C-1   A1XVD200E       
! Controller 376  Bay5    AT-SBx81CFC960                    B-0   A270E400A       
! Controller 376  Bay6    AT-SBx81CFC960                    B-0   A270E400U       
! PSU        319  PSUD    AT-SBxPWRSYS1/AC                  A-0   A045804135000064
! Fan module 375  FAN1    AT-SBxFAN06                       H-0   A04561A141900013
! --------------------------------------------------------------------------------
! --------------------------------------------------------------------------------
! Environment Status : Normal
! Bootloader version : 3.0.5
! Stack member 2
! Board      ID   Bay     Board Name                        Rev   Serial number
! --------------------------------------------------------------------------------
! Chassis    374  Chassis AT-SBx8106                        B-0   R265E3Z0L       
! Blade      381  Bay2    AT-SBx81XS16                      A-0   A04775H131100011
! Blade      380  Bay3    AT-SBx81GT40                      A-1   A26LD700Q       
! Controller 376  Bay5    AT-SBx81CFC960                    B-0   A270E400G       
! Controller 376  Bay6    AT-SBx81CFC960                    C-0   A270F602U       
! PSU        319  PSUD    AT-SBxPWRSYS1/AC                  A-0   A045804134300036
! Fan module 375  FAN1    AT-SBxFAN06                       H-0   A04561A141900010
! --------------------------------------------------------------------------------
! --------------------------------------------------------------------------------
! Environment Status : Normal
! Bootloader version : 3.0.2
! Current boot config: flash:/multi.cfg (file exists)
! System Name
```


Given the controller card can be in either bay 5 and/or 6, I felt it was worthwhile noting which controller card's serial number we're actually displaying in LibreNMS. It also makes it clear it's not the chassis shell's serial. 

It's also worth noting I've only made this code run for this 1 product. 

Thanks,

Matt

Reply:
Also, if there's major concerns I'm adding this to the field, perhaps it's something I could look at removing when I finally get to https://github.com/librenms/librenms/pull/8109.

Since there's no entity mib support at the moment, which serial number we're showing for said chassis product just isn't clear.  However, if the inventory support is in LibreNMS later, then the user of course can look at that tab to clarify further what serial number is in fact being displayed.

Since this isn't an option at the moment, I feel it's worth leaving the change as I currently have it. 

Thoughts?

Thanks,

Matt

Reply:
Ok based on that I think you should remove the text. Their should be no reason to explain what serial number we display just that we pick the one that makes the most sense.

For example on the Cisco ASAs we use the Chassis serial number and I'm sure we do the same elsewhere so it makes more sense to use the chassis serial. The rest of them belong in inventory.

Reply:
>Ok based on that I think you should remove the text. Their should be no reason to explain what serial number we display just that we pick the one that makes the most sense.

OK no worries. Will change in next commit.  

> I'm sure we do the same elsewhere so it makes more sense to use the chassis serial.

Allied Telesis literally ties nothing licensing wise to the Chassis's serial. It's all tied to the controller card[s] in the chassis. Is that the same for Cisco ASAs? I'm assuming ASAs bind it to the Chassis...

I can make the serial the chassis' if you want...I guess I'm reluctant because I can't see any use in knowing the Chassis' serial number other than perhaps Inventory management of physical devices in a network. 

Thanks,

Matt

Reply:
Use the one that makes the most sense :)

Reply:
Done :-) Thanks.